### PR TITLE
Auto expand with multiple hosts

### PR DESCRIPTION
**Pull request by derekcollison**
Allows us to auto-expand hosts that resolve to multiple IPs.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: nats.go [hunk 1]
```diff
@@ -1167,14 +1168,38 @@ func (nc *Conn) createConn() (err error) {
 		cur.lastAttempt = time.Now()
 	}
 
+	// We will auto-expand host names if they resolve to multiple IPs
+	var hosts []string
+	u := nc.current.url
+
+	if net.ParseIP(u.Hostname()) == nil && (nc.Opts.Secure || u.Scheme == tlsScheme) {
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply nats.go]: Good point. Will allow all.

File: nats.go [hunk 2]
```diff
@@ -1167,14 +1168,38 @@ func (nc *Conn) createConn() (err error) {
...
+	}
+
 	// CustomDialer takes precedence. If not set, use Opts.Dialer which
 	// is set to a default *net.Dialer (in Connect()) if not explicitly
 	// set by the user.
 	dialer := nc.Opts.CustomDialer
 	if dialer == nil {
-		dialer = nc.Opts.Dialer
+		// We will copy and shorten the timeout if we have multiple hosts to try.
+		copyDialer := *nc.Opts.Dialer
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply nats.go]: But if they do this will pick it up.

[the remark this answers is not in the document]
Comment (kozlovic) [reply nats.go]: yes.

**Comment by kozlovic** [comment]
LGTM

**Pull request merged**